mondrian/src/main/java/mondrian/rolap/RolapMemberBase.java

```diff
@@ -513,10 +513,16 @@ public Comparable getOrderKey() {
 
     void setOrdinal(int ordinal) {
         if (this.ordinal == -1) {
-            this.ordinal = ordinal;
+          setOrdinal(ordinal, true);
         }
     }
 
+    protected void setOrdinal(int ordinal, boolean forced) {
+      if (forced) {
```

Comment:
This seems weird.  If the caller knows enough to pass forced or not, wouldn't they just be able to not call the method when forced is false?

Reply:
I need the method just to set up ordinal.
In this class we already have the method `void setOrdinal(int ordinal)` with condition:
  `if (this.ordinal == -1)`
This method is using in many places and I don't want to delete this condition because this can affect already existing logic.
That is why I've just overloaded `setOrdinal` method with bool parameter.

Reply:
Changed a bit to not miss `false` case.